doc: add the 'batch' documentation (#49)
AmadeusITGroup/tansu: README.md

* can handle both immutable and mutable data;
* results in compact code with the absolute minimum of boilerplate.

Implementation wise, it is a tiny (500 LOC) library without any external dependencies.

## Comparison with the Svelte stores

Tansu is designed to be and to remain fully compatible with Svelte. Nevertheless, it brings several improvements:

### tansu works well with the Angular ecosystem:
* works with the standard `async` pipe out of the box;
* stores can be registered in the DI container at any level (module or component injector).

### a batch function is available

> Diamond dependency cases can lead to different issues. Let have a look at the following example:

Reviewer: ```suggestion
Depending on multiple stores can lead to some issues. Let's have a look at the following example:
```

The diamond dependency issue is more specific when a store depends on the same ancestor through multiple paths:
```mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
```

In your example, you do not really have a diamond dependency issue, only a dependency on multiple stores that you want to change in an atomic way.

```mermaid
graph TD;
    firstName-->fullName;
    lastName-->fullName;
```

```typescript
import {writable, derived} from 'svelte/store';

const firstName = writable('Arsène');
const lastName = writable('Lupin');
const fullName = derived([firstName, lastName], ([a, b]) => `${a} ${b}`);
fullName.subscribe((name) => console.log(name)); // logs any change to fullName
firstName.set('Sherlock');
lastName.set('Holmes');
console.log('Process end');
```

The output of this example will be:

```
Arsène Lupin
Sherlock Lupin
Sherlock Holmes
Process end
```

The fullName store successively went through different states, including an inconsistent one, as `Sherlock Lupin` does not exist! Even if it can be seen as just an intermediate state, it is **fundamental** for a state management to only manage consistent data in order to prevent issues and optimize the code.

> In Tansu, the batch function is available to differ **synchronously** (another important point) the derived calculation and solve all kind of diamond dependency case.

Reviewer: ```suggestion
In Tansu, the [batch function](https://amadeusitgroup.github.io/tansu/tansu.batch.html) is available to defer **synchronously** (another important point) the derived calculation and solve all kind of multiple dependencies issues.
```

* can handle both immutable and mutable data;
* results in compact code with the absolute minimum of boilerplate.

Implementation wise, it is a tiny (500 LOC) library without any external dependencies.

## Comparison with the Svelte stores

Tansu is designed to be and to remain fully compatible with Svelte. Nevertheless, it brings several improvements:

### tansu works well with the Angular ecosystem:
* works with the standard `async` pipe out of the box;
* stores can be registered in the DI container at any level (module or component injector).

### a batch function is available

Depending on multiple stores can lead to some issues. Let's have a look at the following example:

The diamond dependency issue is more specific when a store depends on the same ancestor through multiple paths:
```mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
```

In your example, you do not really have a diamond dependency issue, only a dependency on multiple stores that you want to change in an atomic way.

```mermaid
graph TD;
    firstName-->fullName;
    lastName-->fullName;

> ```

Reviewer: I don't think you should include this in the README.md, it was only intended to be a comment on the PR, not to be included in the file.
I suppose the mistake was done by GitHub when approving my suggestion of change.

It would make sense to add some graphs in the README.md in another PR to explain those issues that the svelte store has and that are solved by tansu (such as the [asymmetric diamond dependency issue](https://github.com/AmadeusITGroup/tansu/pull/31)).

Author: Well, in fact, the confusion came from me !

I thought you wanted to include the graph to explain the kind of issues it resolves. But right, we can do it in another PR.
